### ISSUE TITLE
Handle mismatch between tabyl and its core after it is arranged

### ISSUE THREAD

**Issue by sfirke** [sfirke/janitor#407]
I could write an arrange method for objects of class `tabyl`.  Implementation might actually be harder than this, but I'm imagining an S3 method that extends `dplyr::arrange()` for a tabyl.  If an object is a two-way tabyl, it sorts the core, too.  It could sort a one-way tabyl's core too, I guess, though I'm not sure it matters.  It might check for totals, print a warning if they're present, sort the core, and then call `dplyr::arrange()`.

Then you could sort a tabyl by something - say, N - and still do `adorn_ns()` with correct results.

**Comment by sfirke** [comment]
I've been thinking that this is a good idea, but with a likelihood of unknown downsides that would only become apparent in user testing.  Thus I hope to add this to the dev version after 2.2 and try it out on GitHub for a while.

**Comment by sfirke** [comment]
Here are some approaches, in the order of how much they appeal to me.

1. *Have `adorn_ns()` do this check for first column match of `core` and input `dat`, if they don't match then sort `core` to match*

Pros: most minimal as it only address the problem at the specific site of the problem, in `adorn_ns()`.  Should it print a notice that it's doing the sort?

2.  *Use `as_tabyl()` to quietly check if rows have gotten sorted and no longer match `core`, if so sort `core`*

Pros: quiet, elegant to have the core stay aligned.  Cons: if user has added a new row, it would get wonky and would need error catching for that, even though user isn't adding Ns..

3. *Write a custom arrange() method for objects of class `tabyl` that arranges the core, too*

Pros: the most robust and elegant.  Cons: may have consequences like Option 2, or more so.  For instance if someone adds a row to a tabyl and then calls arrange, now they have potential problems.  Also, would it print a warning upon package load that it's masking `dplyr::arrange` even though it's not really?

I lean toward option 1 (which I found in a note from 2018, good job writing it down old Sam).  It's not as elegant as an S3 method for `arrange()`, but in practice I think it solves the problem and won't get into any trouble.  As far as I can see, no other `adorn_` function cares about the `core` being out of order, e.g., a `tabyl` -> `arrange` -> `adorn_totals()` works even with the differently-sorted `core.`

**Comment by sfirke** [comment]
Hold on.  I think if I delete these lines https://github.com/sfirke/janitor/blob/main/R/as_and_untabyl.R#L28-L30, then calls to `adorn_totals()` and `adorn_percentages()` will recast the tabyl's `core` in its sorted order.  Those are the only `adorn_` calls to `as_tabyl`.

This doesn't fully solve the problem, because someone might want to calculate row-wise percentages and then sort the tabyl, and then the core would be out of date.  But I might as well remove that block of code while working on Option 1 as I don't *think* it's helping anything and it is contributing to this problem.

**Comment by sfirke** [comment]
Whew, I think I have addressed this pretty well in #464.   Here's the only thing still broken (in a sense) that I can find.  It's separate from what I've patched, just a way to introduce error with an obscure sorting sequence:
```
mtcars %>%
  tabyl(am, cyl) %>%
  adorn_totals() %>%
  arrange(desc(`4`)) %>%
  adorn_percentages("col")

    am         4   6     8
 Total 0.5789474 0.7 0.875
     1 0.4210526 0.3 0.125
     0 0.1578947 0.4 0.750
```
If you add Totals, _then_ sort the totals row up to the top, _then_ calculate col-wise percentages, you get bad results.  But it would be so strange to sort the totals row at that stage, when you could do this instead and get the correct result:

```
mtcars %>%
  tabyl(am, cyl) %>%
  adorn_totals() %>%
  adorn_percentages("col") %>%
  arrange(desc(`4`))

    am         4         6         8
 Total 1.0000000 1.0000000 1.0000000
     1 0.7272727 0.4285714 0.1428571
     0 0.2727273 0.5714286 0.8571429
```

That I'm not going to worry about this.  It falls under the category of applying `adorn_` functions in a sensible order.
